### DISCUSSION TITLE
Is Bearertoken not supported with Twitter v2?

### DISCUSSION THREAD

**Discussion by kamoco** [Xwilarg/TwitterSharp#57] [General]
I am getting unauthorized when to use Bearer token to get any API. one eg: is userinfo. I tried to use the GetUserAsync

**Comment by marcogruhl** [comment]
I just run the tests, and they work. i still have elevated access. The new "Free" tier just provides post/delete tweet and user me endpoints. with the basic tier you should be able to access the most endpoints with this api.
The bearertoken app-based auth is supported with v2 and this api. For user based auth you have to request a token via 3-legged OAuth ([TwitterSharp OAuth Helper](https://github.com/marcogruhl/TwitterSharp/blob/GeneralRequest/src/OAuthHelper.cs)).
